lib/app_home.dart

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright (C) 2022 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import 'package:flutter/material.dart';
+import 'package:yaru_icons/widgets/yaru_icons.dart';
+
+import 'constants.dart';
+import 'installer_state_controller.dart';
+import 'installer_status_widget.dart';
+import 'l10n/app_localizations.dart';
+import 'slide.dart';
+import 'slides_page.dart';
+
+class AppHome extends StatefulWidget {
+  const AppHome({
+    Key? key,
+    required this.slides,
+    required this.controller,
+  }) : super(key: key);
+
+  final List<Slide> slides;
+  final InstallerStateController controller;
+
+  @override
+  State<AppHome> createState() => _AppHomeState();
+}
+
+class _AppHomeState extends State<AppHome> {
+  late final InstallerStateController controller;
```

Comment:
is there any reason to have this extra variable? it's always the same as `widget.controller` so one could use `widget.controller` directly in those few places where the controller is accessed?

Reply:
Good point. Getting rid of the variable also allows getting read of `initState()`.